modules/ppcp-settings/src/Service/PaymentMethodsEligibilityService.php

```diff
@@ -0,0 +1,167 @@
+
+use WooCommerce\PayPalCommerce\Applepay\ApplePayGateway;
+use WooCommerce\PayPalCommerce\Axo\Gateway\AxoGateway;
+use WooCommerce\PayPalCommerce\Googlepay\GooglePayGateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\BancontactGateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\BlikGateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\EPSGateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\IDealGateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\MultibancoGateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\MyBankGateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\P24Gateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\PWCGateway;
+use WooCommerce\PayPalCommerce\LocalAlternativePaymentMethods\TrustlyGateway;
+use WooCommerce\PayPalCommerce\Settings\Data\Definition\FeaturesDefinition;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\CardButtonGateway;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\CreditCardGateway;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\OXXO\OXXO;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\PayUponInvoice\PayUponInvoiceGateway;
+use WooCommerce\PayPalCommerce\WcGateway\Helper\DCCProductStatus;
+
+/**
+ * Manages eligibility checks for various PayPal Commerce features.
+ */
+class PaymentMethodsEligibilityService {
+
+	/**
+	 * PayPal Country (or Woo Country if not onboarded)
+	 */
+	private string $merchant_country;
+
+	/**
+	 * If alternative payment methods are eligible.
+	 */
+	private bool $is_apm_eligible;
+
+	/**
+	 * Array of Merchant capabilities (true: enabled / false: disabled)
+	 */
+	private array $merchant_capabilities;
+
+
+	/**
+	 * ACDC Seller status.
+	 */
+	private DCCProductStatus $dcc_product_status;
+
+	/**
+	 * Whether Axo is eligible.
+	 *
+	 * @var callable
+	 */
+	private $axo_eligible;
+
+	/**
+	 * Whether Card Fields is eligible.
+	 *
+	 * @var callable
+	 */
+	private $card_fields_eligible;
+
+	/**
+	 * Whether Apple Pay is available
+	 */
+	private bool $apple_pay_available;
+
+	/**
+	 * Whether Google Pay is available
+	 */
+	private bool $google_pay_available;
+
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string           $merchant_country PayPal Country (or Woo Country if not onboarded).
+	 * @param bool             $is_apm_eligible If alternative payment methods are eligible.
+	 * @param array            $merchant_capabilities Array of Merchant capabilities (true: enabled / false: disabled).
+	 * @param DCCProductStatus $dcc_product_status ACDC Seller status.
+	 * @param callable         $axo_eligible Whether Axo is eligible.
+	 * @param callable         $card_fields_eligible Whether Card Fields is eligible.
+	 * @param bool             $apple_pay_available Whether Apple Pay is available.
+	 * @param bool             $google_pay_available Whether Google Pay is available.
+	 */
```

Comment:
We no longer need to add constructor and property comments if their purpose is already obvious.
And also I think you have a redundant empty line (50-51)